docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,422 @@
+    description: 'Cool new chart',
+    editorController: 'default',
+    editorConfig: {
+       optionsTemplate: '<my-custom-options-directive></my-custom-options-directive>' // or
+       optionsTemplate: MyReactComponent // or if multiple tabs are required:
+       optionsTabs: [
+           { title: 'tab 1', template: '<div>....</div> },
+           { title: 'tab 2', template: '<my-custom-options-directive></my-custom-options-directive>' },
+           { title: 'tab 3', template: MyReactComponent }
+       ]
+    }
+  }
+-----------
+
+[[development-custom-editor]]
+==== custom editor controller
+You can create a custom editor controller. To do so pass an Editor object (the same format as VisController class). 
+You can make your controller take extra configuration which is passed to the editorConfig property.
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+class MyEditorController {
+    constructor(el, vis) {
+      this.el = el;
+      this.vis = vis;
+      this.config = vis.type.editorConfig;
+   }
+   render(visData) {
+      return new Promise(resolve => {
+         console.log(this.config.my);
+         ...
+         resolve('done rendering');
+      }),
+   destroy() {
+      console.log('destroying');
+   }
+}
+
+const MyNewVisType = (Private) => {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    editorController: MyEditorController,
+    editorConfig: { my: 'custom config' }
+  });
+}
+
+VisTypesRegistryProvider.register(MyNewVisType);
+-----------
+
+[[development-visualization-request-handlers]]
+=== Visualization Request Handlers
+Request handler gets called when one of the following keys on AppState change:
+`vis`, `query`, `filters` or `uiState` and when timepicker is updated. This sums up to pretty much every change to visualization. On top
+of that it will also get called on force refresh.
+
+By default visualizations will use courier request handler. They can also choose to use any of the other provided 
+request handlers. It's also possible to define your own request handler 
+(which you can then register to be used by other visualizations).
+
+[[development-default-request-handler]]
+==== courier request handler
+'courier' is the default request handler which works with the 'default' side bar editor.
+
+[[development-none-request-handler]]
+==== `none` request handler
+Using 'none' as your request handles means your visualization does not require any data to be requested.
+
+[[development-custom-request-handler]]
+==== custom request handler
+You can define your custom request handler by providing a function with the following definition:
+`function (vis, appState, uiState, searchSource) { ... }`
+
+Function must return a promise, which should get resolved with new data that will be passed to responseHandler.
```

Comment:
This `function` must ...

Reply:
time for breakfast .... arrrrgh prepositions ... :)